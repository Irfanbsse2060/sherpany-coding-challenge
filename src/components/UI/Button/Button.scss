@import "../../../styles/variables";

.custom-button {
  display: inline-block;
  margin-bottom: 0;
  margin-top: 0;
  font-weight: 500;
  text-align: center;
  vertical-align: middle;
  white-space: nowrap;
  transition: 0.5s;
  padding: 5px 12px;
  border: 2px solid;
  border-radius: 20px;
  line-height: 1;
  min-width: 85px;
  cursor: pointer;

  &:hover {
    background: transparent;
    color: $scorpion;
  }

  &:focus {
    outline: none;
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }

  &--primary {
    background: $primary-button-accent;
    color: $white;
    border-color: $primary-button-accent;
  }

  &--secondary {
    background: $secondary-button-accent;
    color: $black;
    border-color: $secondary-button-accent;
  }
}
