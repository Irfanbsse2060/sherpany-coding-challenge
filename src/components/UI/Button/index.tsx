import React, {MouseEvent} from 'react'
import './Button.scss'

export enum ButtonType { primary, secondary}

interface TextFieldProps {
    text: string
    type?: ButtonType
    handleClick: (event: MouseEvent<HTMLButtonElement>) => void
    testId?: string,
    style?: string
}

export default function Button(props: TextFieldProps) {
    const {type = ButtonType.primary, testId = 'custom-button', style = {}, text, handleClick} = props
    const buttonType = type === ButtonType.primary ? 'primary' : 'secondary'


    return (
        <button style={style} data-testid={testId}
                className={`custom-button custom-button--${buttonType}`}
                onClick={handleClick}>
            {text}
        </button>
    )

}
