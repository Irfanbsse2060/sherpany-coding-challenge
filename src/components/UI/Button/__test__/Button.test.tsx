import React from 'react';
import {fireEvent, render} from '@testing-library/react';
import Button, {ButtonType} from '../index'

describe('Button', () => {

    test('should render primary type', () => {
        const {container} = render(<Button text='text'  handleClick={jest.fn()}/>);
        expect(container).toMatchSnapshot()
    })

    test('should render secondary type', () => {
        const {container} = render(<Button text='text' type={ButtonType.secondary} handleClick={jest.fn()}/>);
        expect(container).toMatchSnapshot()
    })

    test('should call handle click method on click', () => {
        const mockHandleChange = jest.fn()
        const {getByTestId} = render(<Button text='text' handleClick={mockHandleChange}/>);
        fireEvent.click(getByTestId('custom-button'))
        expect(mockHandleChange).toHaveBeenCalledTimes(1)
    });

})
