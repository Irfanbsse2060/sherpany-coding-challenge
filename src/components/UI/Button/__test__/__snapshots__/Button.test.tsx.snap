// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render primary type 1`] = `
<div>
  <button
    class="custom-button custom-button--primary"
    data-testid="custom-button"
  >
    text
  </button>
</div>
`;

exports[`Button should render secondary type 1`] = `
<div>
  <button
    class="custom-button custom-button--secondary"
    data-testid="custom-button"
  >
    text
  </button>
</div>
`;
